### PR TITLE
test: Add GraphQL execution engine robustness tests

### PR DESCRIPTION
**Pull request by pekkah**
## Summary
- Comprehensive execution engine robustness tests covering edge cases and error conditions
- Tests null/empty queries, syntax errors, validation errors, deeply nested queries, circular fragments  
- Tests resolver error handling, variable validation, directive validation, fragment validation
- Tests multiple operation handling and type validation scenarios
- Covers performance scenarios with large queries and slow resolvers

## Test plan
- [x] Builds successfully with proper ResolversMap syntax
- [x] Uses correct Executor.Execute API patterns  
- [x] Follows existing test conventions and patterns
- [ ] All test cases pass (some test expectations may need adjustment based on actual API behavior)
- [ ] Tests provide good coverage of execution engine edge cases

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: tests/GraphQL.Tests/ExecutionEngineRobustnessFacts.cs [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+        {
+            query += $"alias{i}: hello ";
+        }
+        query += "}";
+
+        var result = await Executor.Execute(schema, query);
+
+        Assert.NotNull(result);
+        // Should complete within reasonable time and memory constraints
+    }
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/ExecutionEngineRobustnessFacts.cs]: ✅ Fixed: Added concrete assertions to verify query execution success and replaced string concatenation with StringBuilder. The test now verifies that the large query executes without errors and returns proper data, making it deterministic and measurable.

File: tests/GraphQL.Tests/ExecutionEngineRobustnessFacts.cs [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+    [Fact]
+    public async Task Execute_WithResolverTimeout_ShouldHandleGracefully()
+    {
+        var schema = await CreateSchemaWithSlowResolver();
+
+        var result = await Executor.Execute(schema, "{ slowField }");
+
+        Assert.NotNull(result);
+        // Should either complete or timeout gracefully
+    }
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/ExecutionEngineRobustnessFacts.cs]: ✅ Fixed: Added specific assertions to verify that the slow resolver completes successfully and returns the expected data. The test now checks for Assert.Null(result.Errors), Assert.NotNull(result.Data), and Assert.Equal("Slow result", result.Data["slowField"]) to ensure deterministic behavior.
